### ISSUE TITLE
Hypertables expand add_dimension() secondary dimension with List partitioning instead of Hash

### ISSUE THREAD

**Issue by muntdan** [timescale/timescaledb#6179]
### What type of enhancement is this?

API improvement

### What subsystems and features will be improved?

Adaptive chunking

### What does the enhancement do?

Today there is support for secondary dimension only as a hash: https://docs.timescale.com/api/latest/hypertable/add_dimension/
This means you cant have full control on a Multi - Tenant scenario that distinct values dont end up in the same chunk.
Expanding the capability/parameters of this function to accept a List of static values and create exact LIST partitioning towards them would solve this.

**Comment by adriangb** [comment]
This would be a really useful feature for us. Especially if support for retention policies and tiered storage was expanded to support the additional dimensions. It would be a standout feature that competitors simply don’t offer.

**Comment by ewoolsey** [comment]
Yeah agreed. This would be a killer feature. Api would likely look like this:

```sql
SELECT add_dimension('my_table', by_list('my_column'));
```
